src/core/coerce.c

```diff
@@ -378,6 +378,42 @@ void MVM_coerce_smart_numify(MVMThreadContext *tc, MVMObject *obj, MVMRegister *
         else
             MVM_exception_throw_adhoc(tc, "cannot numify this");
     }
+    {
+        MVMint64 t = (MVMint64)res_reg->n64;
+        if ((MVMnum64)t != res_reg->n64) {
+            //fprintf(stderr, "got an actual non-int value %g\n", res_reg->n64);
+        }
+    }
```

Comment:
This block looks like a dead code debug leftover; I must have missed it in the first review, sorry.

Reply:
Argh, can't believe I committed that. Anyway, force pushed another update.